src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php

```diff
@@ -163,8 +164,50 @@ private function collectOnClient(TraceableHttpClient $client): array
             unset($traces[$i]['info']); // break PHP reference used by TraceableHttpClient
             $traces[$i]['info'] = $this->cloneVar($info);
             $traces[$i]['options'] = $this->cloneVar($trace['options']);
+            $traces[$i]['curlCommand'] = $this->getCurlCommand($trace);
         }
 
         return [$errorCount, $traces];
     }
+
+    private function getCurlCommand(array $trace): string
+    {
+        $method = $trace['method'];
+        $options = $trace['options'];
```

Comment:
should it be based on options or on the `info` reported by the client in the response ? The later will account for all request headers added by the wrapped client too, while options will not (due to the `defaultOptions` feature)

Reply:
`$trace['info']` does not seem to have any way to get the request header. Should I parse the `debug`key inside? :cold_sweat: 
![Screenshot of a typical trace](https://user-images.githubusercontent.com/7600265/140496140-fbc23f02-3d27-4e10-8230-26fa64c6ce16.png)

Reply:
@nicolas-grekas I remember that you told me that `$response->getInfo()` allows to access that information. Was it wrong ?

Reply:
I confirm request headers are not in the info, except in the debug stream. Parsing it should be the way to go.

Reply:
Maybe we should add the full request headers in the info, instead of relying on parsing (AFAIK, the format of the debug stream is implementation-specific)

Reply:
It'd prefer going with parsing for now. The format is borrowed from curl and it's the same in all our implems, and it's very easy to parse.

Reply:
Leaving this conversation unresolved for now, just want to make sure I have covered all the cases before :)